cmd/cody-gateway/internal/actor/actor.go

```diff
@@ -222,24 +222,35 @@ func (a *Actor) Limiter(
 	feature codygateway.Feature,
 	rateLimitNotifier notify.RateLimitNotifier,
 ) (limiter.Limiter, bool) {
+	// Start with the base limiter. For most users, this is governed by the app's configuration.
+	// But for users on the Cody Pro plan, their limits may be different.
```

Comment:
Does this comment apply at this level? We should avoid anything Cody-Pro specific as this is the shared Actor abstraction

Reply:
It's possible the comment is totally wrong and I don't fully understand the code. But if it is factually correct, I don't think _mentioning_ Cody Pro would be a problem.

I agree that we don't want Cody Gateway to have any awareness of "Cody Pro" or "Cody Free" etc. And that any rate limits are just opaque values that come from the Actor source. (i.e. it is dotcom that knows if someone is on Cody-Pro or not, and then returns different values for Actor.RateLimits, etc.)

But as far as mentioning it in comments like here, IMHO it's helpful to call out why we have the base limiter and/or why rate limits would need to be refreshed at different intervals for different users.

Reply:
Reading this again, I find it pretty confusing IMO - a better one would be:

```suggestion
	// Start with the base limiter. This limiter enforces rate limits provided
	// the actor's Source, and each Source implementation owns ensuring that
	// Cody Gateway has an up-to-date view of the appropriate rate limits for
	// a particular actor based on the application (e.g. users from enterprise product
	// subscriptions, Self-Serve-Cody's tiers, etc.
```

Reply:
The comment makes it sound like we have 2 particular type of limiters and rate limits, when in reality, Cody Gateway just wants to enforce whatever rate limits the Source of an Actor says the Actor has. It doesn't care about plans or why limits are different or why they change

Reply:
I believe we have both participated in [Cunningham's Law](https://en.wikipedia.org/wiki/Ward_Cunningham#%22Cunningham's_Law%22)-driven development. 🙃😭